services/user/Permissions/ui/src/permissions/App.tsx

```diff
@@ -1,83 +1,111 @@
-import { useEffect, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 
 import { Button } from "@shadcn/button";
-import { Checkbox } from "@shadcn/checkbox";
 
-import { siblingUrl } from "@psibase/common-lib";
-import { Nav } from "@components/nav";
+import { getSupervisor, siblingUrl } from "@psibase/common-lib";
+import { ActivePermsOauthRequest } from "./db";
 
-import { supervisor } from "./perms_main";
+const supervisor = getSupervisor();
 
 export const App = () => {
     const thisServiceName = "permissions";
-    const [isLoading, setIsLoading] = useState<boolean>(true);
-    const [params, setParams] = useState<any>({});
-    const [isValidPermRequest, setIsValidPermRequest] = useState<any>({});
-
-    const init = async () => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [validPermRequest, setValidPermRequest] = useState<any>(null);
+    const [error, setError] = useState<string | null>(null);
+    const initApp = useCallback(async () => {
         await supervisor.onLoaded();
```

Comment:
awaiting `onLoaded` is no longer necessary either. Anything that calls the supervisor impl internally already does the await onLoaded